src/Microsoft.Extensions.Diagnostics.HealthChecks/IHealthCheckService.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Diagnostics.HealthChecks
+{
+    /// <summary>
+    /// A service which can be used to check the status of <see cref="IHealthCheck"/> instances registered in the application
+    /// </summary>
+    public interface IHealthCheckService
+    {
+        /// <summary>
+        /// A <see cref="IReadOnlyDictionary{TKey, T}"/> containing all the health checks registered in the application.
+        /// </summary>
+        /// <remarks>
+        /// The key maps to the <see cref="IHealthCheck.Name"/> property of the health check, and the value is the <see cref="IHealthCheck"/>
+        /// instance itself.
+        /// </remarks>
+        IReadOnlyDictionary<string, IHealthCheck> Checks { get; }
```

Comment:
Why does this need to be exposed publicly? Only the service itself consumes them, no?

Reply:
Has there been any discussion on the ability to add and remove checks at runtime?

Reply:
Callers may want to run individual health checks or groups of health checks. The previous system had methods and infrastructure for this, but we didn't want to add that complexity in the full product. However, we still want consumers to be able to access and run individual checks.

Reply:
> Has there been any discussion on the ability to add and remove checks at runtime?

No, and I don't think that's a feature we've seen in any other systems. It's not really the kind of thing you'd want to add after the app starts.

Reply:
We get quite a few requests to be able to add things without app restarts. E.g. middleware, auth handlers, etc..

Reply:
We'll wait for those requests then :)